src/Bids/Woolsey.hs

```diff
@@ -0,0 +1,221 @@
+    makeCall $ T.Bid 2 T.Hearts
+
+
+b1No2C2S :: Action
+b1No2C2S = nameAction "wool_b1No2C2S" $ do
+    -- If your major suits are the same length, bid the better one. That's
+    -- slightly hard to code up, so just avoid those situations entirely.
+    T.Spades `longerThan` T.Hearts
+    makeCall $ T.Bid 2 T.Spades
+
+
+b1No2D :: Action
+b1No2D = nameAction "wool_b1No2D" $ do
+    alternatives [b1No2D2HP, b1No2D2H2S]
+    makeAlertableCall (T.Bid 2 T.Diamonds) "one long major"
+
+
+b1No2D2H :: Action
+b1No2D2H = nameAction "wool_b1No2D2H" $ do
+    -- If you've got your own long suit, you might bid it. To prevent users from
+    -- being tempted to do that, make sure there aren't long suits.
+    forEach T.allSuits (`maxSuitLength` 5)
+    makeAlertableCall (T.Bid 2 T.Hearts) "pass or correct"
+
+
+b1No2D2HP :: Action
+b1No2D2HP = nameAction "wool_b1No2D2HP" $ do
+    singleSuited T.Hearts
+    makeCall T.Pass
+
+
+b1No2D2H2S :: Action
+b1No2D2H2S = nameAction "wool_b1No2D2H2S" $ do
+    singleSuited T.Spades
+    makeCall $ T.Bid 2 T.Spades
+
+
+b1No2H :: Action
+b1No2H = nameAction "wool_b1No2H" $ do
+    forEach T.minorSuits (\m ->
+        twoSuited T.Hearts m >> minSuitLength T.Hearts 5)
+    makeAlertableCall (T.Bid 2 T.Hearts) "5+ hearts and 4+ in a minor"
+
+
+b1No2S :: Action
+b1No2S = nameAction "wool_b1No2S" $ do
+    forEach T.minorSuits (\m ->
+        twoSuited T.Spades m >> minSuitLength T.Spades 5)
+    makeAlertableCall (T.Bid 2 T.Spades) "5+ spades and 4+ in a minor"
+
+
+preferMinor_ :: T.Suit -> Action
+preferMinor_ major = do
+    -- There are lots of other hand shapes that would prefer the minor, but
+    -- figuring them all out is harder than just limiting the hand types to the
+    -- easy ones. When you encounter the hard ones at the table, hopefully this
+    -- practice has helped.
+    maxSuitLength major 2
+    forEach T.minorSuits (`minSuitLength` 4)
+    makeAlertableCall (T.Bid 2 T.Notrump) "bid your minor"
+
+b1No2H2N :: Action
+b1No2H2N = nameAction "wool_b1No2H2N" (preferMinor_ T.Hearts)
+
+b1No2S2N :: Action
+b1No2S2N = nameAction "wool_b1No2S2N" (preferMinor_ T.Spades)
+
+
+b1No2H2N3C :: Action
+b1No2H2N3C = nameAction "wool_b1No2H2N3C" $ do
+    T.Clubs `longerThan` T.Diamonds
+    makeCall (T.Bid 3 T.Clubs)
+
+
+b1No2H2N3D :: Action
+b1No2H2N3D = nameAction "wool_b1No2H2N3D" $ do
+    T.Diamonds `longerThan` T.Clubs
+    makeCall (T.Bid 3 T.Diamonds)
+
+
+b1No2S2N3C :: Action
+b1No2S2N3C = nameAction "wool_b1No2S2N3C" $ do
+    T.Clubs `longerThan` T.Diamonds
+    makeCall (T.Bid 3 T.Clubs)
+
+
+b1No2S2N3D :: Action
+b1No2S2N3D = nameAction "wool_b1No2S2N3D" $ do
+    T.Diamonds `longerThan` T.Clubs
+    makeCall (T.Bid 3 T.Diamonds)
+
+
+b1No2N :: Action
+b1No2N = nameAction "wool_b1No2N" $ do
+    twoSuited T.Clubs T.Diamonds
+    makeAlertableCall (T.Bid 2 T.Notrump) "both minors"
+
+
+b1No2N3C :: Action
+b1No2N3C = nameAction "wool_b1No2N3C" $ do
+    T.Clubs `atLeastAsLong` T.Diamonds
```

Comment:
This isn't right. I should probably make something in EDSL.hs trying to describe one suit as stronger than another.